typings/index.d.ts

```diff
@@ -0,0 +1,277 @@
+import * as React from 'react';
+
+type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
+
+interface AnyObject {
+  [key: string]: any;
+}
+
+interface CommonProps<T = any> extends AnyObject {
+  instanceRef?: (instance: React.Ref<T>) => void;
+  onLoad?: (ymaps: YMapsApi) => void;
+  onError?: (error: Error) => void;
+  modules?: string[];
+  children?: React.ReactNode;
+}
+
+export type MapType = 'yandex#map' | 'yandex#satellite' | 'yandex#hybrid';
+
+export interface YMapsApi extends AnyObject {}
+
+//
+// Main components
+// ----------------------------------------------------------------------
+
+export interface YMapsProps extends AnyObject {
+  version?: string;
+  enterprise?: boolean;
+  query?: {
+    lang?: 'tr_TR' | 'en_US' | 'en_RU' | 'ru_RU' | 'ru_UA' | 'uk_UA';
+    apikey?: string;
+    coordorder?: 'latlong' | 'longlat';
+    load?: string;
+    mode?: 'release' | 'debug';
+    csp?: boolean;
+    ns?: string;
+  };
+  children?: React.ReactNode;
+  preload?: boolean;
+}
+
+interface MapStateBase extends AnyObject {
+  controls?: string[];
+  behaviors?: string[];
+  margin?: number[] | number[][];
+  type?: MapType;
+}
+
+interface MapStateBounds {
+  bounds: number[][];
+  zoom?: never;
+  center?: never;
+}
+
+interface MapStateCenter {
+  center: number[];
+  zoom: number;
+  bounds?: never;
+}
+
+/**
+ * REVIEW: discuss if this behavior is necessary
```

Comment:
I think it's a great addition! It's impossible to express those with propTypes and runtime validations really complicate the codebase